ocaml/idl/datamodel_vtpm.ml

```diff
@@ -16,17 +16,47 @@ open Datamodel_types
 open Datamodel_common
 open Datamodel_roles
 
+let get_contents =
+  call ~name:"get_contents" ~in_product_since:"rel_next"
+    ~doc:"Obtain the contents of the TPM" ~secret:true
+    ~params:[(Ref _vtpm, "self", "The VTPM reference")]
+    ~result:(String, "The contents") ~hide_from_docs:true
```

Comment:
We've got a Base64 type in the RPC, can we try using that, would it help? This is a binary string but we want to avoid storing binary strings (for varstored we store it as base64).

Reply:
Yes, I think it can help, does the database do the encoding and decoding or should the functions themselves do it?

Reply:
I don't think the variant support base64: https://github.com/xapi-project/xen-api/blob/master/ocaml/idl/datamodel_types.ml#L390
The base64 encoding is available in the RPC interface.

I found that the license uses base64encoding as well, but the conversion is forced upon the user. This could be an option if there are usecase that need to keep the vtpm contents base64-encoded

Reply:
The contents are now encoded and decoded: both functions use exceptions to report errors, decoding shouldn't be a problem since we always do the encoding.

Regarding encoding I don't see how the code would throw an exception as I can't see the internal function returning an error.

This of course means the vtpm are always encoded/decoded. If this operation are common enough this might become a performance bottleneck. We can change this afterwards and do the encoding/deciding in the clients if we need more flexibility.

Reply:
Ah we might need to do the base64 encoding in the clients anyway: you can't have binary data inside JSON/XML.

Reply:
For now lets keep the base64 encoding here and see what happens with the client, if it cannot send binary data through the wire protocl we'll need to move the base64 encoding there